### PR TITLE
Add RequestCookies class to collect HTTP cookies

### PR DESCRIPTION
**Pull request by hacan359**
- Created the RequestCookies class to manage cookies.
- It gets cookies from a RequestProviderInterface.
- Added methods to retrieve a cookie by name and check if a cookie exists.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 1.1.0 under development
 
 - Enh #10: Add `yiisoft/middleware-dispatcher` event listener to keep `Request` always available (@xepozz)
+- Add `Yiisoft/RequestProvider/RequestCookies` request cookie collection object (@hacan359)
```
Comment (vjik) [inline CHANGELOG.md:6]:
```suggestion
- Add `RequestCookies` class that provides convenient access to request cookies (@hacan359)
```

File: tests/RequestCookiesTest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use Yiisoft\RequestProvider\RequestCookies;
+use Yiisoft\RequestProvider\RequestProviderInterface;
+
+use function dirname;
+
+final class RequestCookiesTest extends TestCase
+{
+    public function testRequestCookiesGet(): void
+    {
+        $requestCoockies = $this->getRequestCookies();
```
Comment (vjik) [inline tests/RequestCookiesTest.php:20]:
```suggestion
        $requestCookies = $this->getRequestCookies();
```

File: tests/RequestCookiesTest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+use Yiisoft\Di\Container;
+use Yiisoft\Di\ContainerConfig;
+use Yiisoft\RequestProvider\RequestCookies;
+use Yiisoft\RequestProvider\RequestProviderInterface;
+
+use function dirname;
+
+final class RequestCookiesTest extends TestCase
+{
+    public function testRequestCookiesGet(): void
```
Comment (vjik) [inline tests/RequestCookiesTest.php:18]:
```suggestion
    public function testGet(): void
```

File: tests/RequestCookiesTest.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $requestCoockies = $this->getRequestCookies();
+
+        $this->assertSame('value', $requestCoockies->get('test'));
+    }
+
+    public function testRequestCookiesHas(): void
```
Comment (vjik) [inline tests/RequestCookiesTest.php:25]:
```suggestion
    public function testHas(): void
```

File: tests/RequestCookiesTest.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $requestCoockies = $this->getRequestCookies();
```
Comment (vjik) [inline tests/RequestCookiesTest.php:27]:
```suggestion
        $requestCookies = $this->getRequestCookies();
```

File: tests/RequestCookiesTest.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+        $this->assertTrue($requestCoockies->has('test'));
+
+        $this->assertFalse($requestCoockies->has('value'));
+    }
+
+    private function getRequestCookies(): RequestCookies
+    {
+        $container = $this->createContainer();
```
Comment (vjik) [inline tests/RequestCookiesTest.php:36]: Here we test `RequestCookies` only, so we can don't use container and create `RequestProvider` directly.

**Review by vjik — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
Thanks for the pull request. I wonder what is the benefit compared to current master:

```php
public function actionX(RequestProviderInterface $requestProvider)
{
    $cookies = $requestProvider->get()->getCookieParams();
	if (!empty($cookies['x'])) {
        $x = $cookies['x'];
        // ...
    }
}
```

To me, new code looks quite similar:

```php
public function actionX(RequestCookies $requestCookies)
{
    $cookies = $requestProvider->get()->getCookieParams();
	if ($requestCookies->has('x')) {
        $x = $requestCookies->get('x');
        // ...
    }
}
```

**Comment by vjik** [comment]
> To me, new code looks quite similar:

New code is simpler and semantically more beautiful.

File: README.md [hunk 7]
```diff
@@ -54,6 +54,23 @@ final class MyService
 }
 ```
 
+### Request cookies collection
+
+You can work with cookie request from any place in you project. Add in config you app in middleware block
```
Comment (samdark) [inline README.md:59]:
```suggestion
You can work with cookies like the following:
```

File: README.md [hunk 8]
```diff
@@ -54,6 +54,23 @@ final class MyService
...
+
+```php
+class MyClass {
```
Comment (samdark) [inline README.md:62]:
```suggestion
class MyClass
{
```

File: README.md [hunk 9]
```diff
@@ -54,6 +54,23 @@ final class MyService
...
+  public function __construct(
+    private \Yiisoft\RequestProvider\RequestCookies $cookies
+  ) {}
+  
+  public function go(): void {
```
Comment (samdark) [inline README.md:67]:
```suggestion
  public function go(): void
  {
```

File: src/RequestCookies.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public function get(string $name): ?string
+    {
+        return $this->cookies[$name] ?? null;
+    }
+
+    public function has(string $name): bool
+    {
+        return isset($this->cookies[$name]);
```
Comment (samdark) [inline src/RequestCookies.php:29]:
```suggestion
        return array_key_exists($name, $this->cookies);
```

**Review by samdark — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (hacan359) [reply CHANGELOG.md]: done

File: tests/RequestCookiesTest.php (hunk 2 above)
Comment (hacan359) [reply tests/RequestCookiesTest.php]: done

File: tests/RequestCookiesTest.php (hunk 3 above)
Comment (hacan359) [reply tests/RequestCookiesTest.php]: done

File: tests/RequestCookiesTest.php (hunk 4 above)
Comment (hacan359) [reply tests/RequestCookiesTest.php]: done

File: tests/RequestCookiesTest.php (hunk 5 above)
Comment (hacan359) [reply tests/RequestCookiesTest.php]: done

File: README.md (hunk 7 above)
Comment (hacan359) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (hacan359) [reply README.md]: done

File: README.md (hunk 9 above)
Comment (hacan359) [reply README.md]: done

File: src/RequestCookies.php (hunk 10 above)
Comment (hacan359) [reply src/RequestCookies.php]: done

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,7 @@
 ## 1.1.0 under development
 
 - Enh #10: Add `yiisoft/middleware-dispatcher` event listener to keep `Request` always available (@xepozz)
+- Add `RequestCookies` class that provides convenient access to request cookies (@hacan359)
```
Comment (samdark) [inline CHANGELOG.md:6]:
```suggestion
- New #11: Add `RequestCookies` class that provides convenient access to request cookies (@hacan359)
```

**Review by vjik — APPROVED** [review]

**Comment by vjik** [comment]
@hacan359 Thank you 👍

**Pull request merged**